testsuite/src/main/java/io/netty/testsuite/transport/socket/SocketSslEchoTest.java

```diff
@@ -523,6 +516,19 @@ public final void exceptionCaught(ChannelHandlerContext ctx, Throwable cause) th
             super(recvCounter, negoCounter, exception);
         }
 
+        @Override
+        public void handlerAdded(final ChannelHandlerContext ctx) {
+            if (!autoRead) {
+                ctx.pipeline().get(SslHandler.class).handshakeFuture().addListener(
+                        new GenericFutureListener<Future<? super Channel>>() {
```

Comment:
```suggestion
                        new ChannelFutureListener() {
```

Reply:
It's not a listener for a `ChannelFuture` but rather for a `Future` that result in a `Channel`. The `ChannelFutures` result in `Void`.

Reply:
Ah, you're right. :sweat_smile: